Handle SIGINT when running "up" command to shutdown gracefully

mokazemi opened the pull request:
This pull request enhances the podman compose behavior by handling the` Ctrl+C` (SIGINT) signal more gracefully. When `Ctrl+C` is pressed during the execution of podman compose up, it runs the `down` command to make sure containers shut down gracefully.

**Solves:** #51, #112, #411, #436, #457, #543, #676

p12tic commented:
Thanks for the contribution. The PR looks good in principle, I'm just not confident enough with the `down_args = argparse.Namespace(**dict(args.__dict__, volumes=False))` line yet.

Separately, pylint is complaining in https://github.com/containers/podman-compose/actions/runs/13497452404/job/37707798339?pr=1148. I think both warnings can be disabled in .pylintrc.

mokazemi commented:
> Thanks for the contribution. The PR looks good in principle, I'm just not confident enough with the `down_args = argparse.Namespace(**dict(args.__dict__, volumes=False))` line yet.
> 

About the line that you mentioned (down arguments) I actually used the same line already used somewhere else in the code:
https://github.com/containers/podman-compose/blob/bcce413af196ece0b8a125f81b14c29bb306ebf0/podman_compose.py#L2742

I guess it should work enough, but we can modify in case of improvements. 

>
> Separately, pylint is complaining in https://github.com/containers/podman-compose/actions/runs/13497452404/job/37707798339?pr=1148. I think both warnings can be disabled in .pylintrc.

Oh I didn't check for warning here. I changed the code about the second one so that it passes now, but for the first one, I added it to `.pylintrc` since IMO it could be ignored :)).

pablo-tx commented:
@p12tic is there anything else blocking this merge?

p12tic commented:
@pablo-tx Sorry for taking a long time to reply.

Code changes look good. Please rebase on top of latest main branch and fix merge conflicts. Also please add a release note to the [newsfragments](https://github.com/containers/podman-compose/tree/main/newsfragments) directory (you can look [here](https://github.com/containers/podman-compose/commit/6de335beb295d8111f52053d75c4051f9f89b7be) for inspiration on how release note looks like). Also DCO test fails - it asks for `Signed-off-by: Author Name <authoremail@example.com>` line in commit message.

p12tic commented:
@pablo-tx @mokazemi  Sorry for taking a long time to reply.

Code changes look good. Please rebase on top of latest main branch and fix merge conflicts. Also please add a release note to the [newsfragments](https://github.com/containers/podman-compose/tree/main/newsfragments) directory (you can look [here](https://github.com/containers/podman-compose/commit/6de335beb295d8111f52053d75c4051f9f89b7be) for inspiration on how release note looks like). Also DCO test fails - it asks for `Signed-off-by: Author Name <authoremail@example.com>` line in commit message.
